### PR TITLE
Add search and profile functionality

### LINKED ISSUE #10: a user can search & join a community

### LINKED ISSUE #11: a user can search for an event

### LINKED ISSUE #12: a user can see events on their profile
events that they are indicated as the ones they are going to/communities they are involved in

### PR DESCRIPTION
**Pull request by robo360**
- Added search functionality
- Added profile functionality

### CODE REVIEW

File: app/src/main/java/com/example/game/fragments/CommunityProfileFragment.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import android.util.Log;
+
+import com.example.game.R;
+import com.example.game.models.Community;
+import com.example.game.models.Subscription;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+
+public class CommunityProfileFragment extends CommunitySearchFragment {
+    private static final String TAG = "CommunityPpFragment";
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/CommunityProfileFragment.java:12]: typo?

File: app/src/main/java/com/example/game/fragments/EventDetailFragment.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
     public static final String KEY_EVENT = "event";
     public static final String KEY_COMMUNITY = "community";
 
-    private Event event;
+    @Nullable private Event event;
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/EventDetailFragment.java:38]: 👍

File: app/src/main/java/com/example/game/fragments/SearchFragment.java [hunk 3]
```diff
@@ -39,15 +37,19 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         SearchView searchView = binding.searchView;
         btnEvent = binding.btnFilter;
         btnCommunity = binding.btnCommunity;
-        searchFragmentManager = getActivity().getSupportFragmentManager();
+        searchFragmentManager = Objects.requireNonNull(getActivity()).getSupportFragmentManager();
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/fragments/SearchFragment.java:40]: 👍

**Review by LynnHX — COMMENTED** [review]
lgtm 😄

File: app/src/main/java/com/example/game/fragments/CommunityProfileFragment.java (hunk 1 above)
Comment (robo360) [reply app/src/main/java/com/example/game/fragments/CommunityProfileFragment.java:12]: No, I had a longer tag so I shortened. It was more than the required 23 characters.

**Review by LynnHX — APPROVED** [review]

**Pull request merged**